### PR TITLE
Replace TableProxyModel with TableDelegate.

### PR DESCRIPTION
**Pull request by spxtr**
This fixes the ugly spaces around the pixmaps in the table view, and is also logically simpler than using the TableProxyModel.

There are still a few outstanding issues with the table view. It doesn't scroll properly, certain columns might not resize properly (not sure), and it doesn't share its selection with the list view.

[Review on Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3633)

### CODE REVIEW

File: Source/Core/DolphinQt2/GameList/GameList.h [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 #include "DolphinQt2/GameList/GameFile.h"
 #include "DolphinQt2/GameList/GameListModel.h"
+#include "DolphinQt2/GameList/TableDelegate.h"
```
Comment (lioncash) [inline Source/Core/DolphinQt2/GameList/GameList.h]: forward declaration

**Comment by lioncash** [comment]
LGTM aside from that

File: Source/Core/DolphinQt2/GameList/GameList.h (hunk 1 above)
Comment (spxtr) [inline Source/Core/DolphinQt2/GameList/GameList.h]: Done. I should make a separate PR to make lots of other things forwardly-declared, since I haven't been.
